aws/resource_aws_iam_role_test.go

```diff
@@ -399,6 +412,41 @@ func testAccAddAwsIAMRolePolicy(n string) resource.TestCheckFunc {
 	}
 }
 
+func testAccCheckAWSRolePermissionsBoundary(n string, expectedPermissionsBoundaryArn string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Role name is set")
+		}
+
+		iamconn := testAccProvider.Meta().(*AWSClient).iamconn
+
+		resp, err := iamconn.GetRole(&iam.GetRoleInput{
```

Comment:
Instead of making a new API call here, you can just pass in `&role` instead of `resourceName` (which is the same `iam.GetRoleOutput` of `resp`). `testAccCheckAWSRoleExists()` does all the above logic already. 😄

Reply:
Thank you for the feedback @bflad !
Fixed it, indeed it's much simpler that way :)